### PR TITLE
PMM-9951 Adding Support for PMM + PGSM Integration tests on PR's

### PR DESCRIPTION
**Pull request by puneet0191**

### CODE REVIEW

**Comment by Naeem-Akhter** [comment]
As discussed in the meeting (PMM+PGMS integration) and discussed in slack, PG 11 & 12 jobs in this PR are failing due to different columns names in PGSM for mean_exec_time (PG 13 &14) --> mean_time (PG 11 &12) , and total_exec_time (PG 13 &14) --> total_time (PG 11 &12). This needs to be handled on PMM side, I would suggest to wait for Jiri input/confirmation before make any progress.

**Comment by Naeem-Akhter** [comment]
Closing this PR, as it needs a fix on PMM side. Issue needs to be fixed by Jiri (PMM dev) we can be merged. 

However for development and testing purposes to make some progress on other stuff , you can create new PRs for testing or can clone your own repo to test the same. I will connect with you regrading it.

**Review by Naeem-Akhter — COMMENTED** [review]
As discussed in the meeting (PMM+PGMS integration) and discussed in slack, PG 11 & 12 jobs in this PR are failing due to different columns names in PGSM for mean_exec_time (PG 13 &14) --> mean_time (PG 11 &12) , and total_exec_time (PG 13 &14) --> total_time (PG 11 &12). This needs to be handled on PMM side, I would suggest to wait for Jiri input/confirmation before make any progress.

Closing this PR, as it needs a fix on PMM side. Issue needs to be fixed by Jiri (PMM dev) we can be merged.

However for development and testing purposes to make some progress on other stuff , you can create new PRs for testing or can clone your own repo to test the same. I will connect with you regrading it.

**Review by capri1989 — CHANGES_REQUESTED** [review]

**Comment by JiriCtvrtka** [comment]
Commits are squashed during merge here on github. I am not sure if this working same in this repo.

**Comment by JiriCtvrtka** [comment]
I see, in repos of core team we have button "merge and squash" in github and you need only click on it, everything else is done by github (merge and squash). So no extra effort, maybe we can configure this repo in same way? What do you think @capri1989 ?

File: .github/workflows/postgresql-11-pmm.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: postgresql-11-pmm-integration
+on: pull_request
+
+jobs:
+  build:
+    name: pg-11-pgsm-pmm-integration-test
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (puneet0191) [reply .github/workflows/postgresql-11-pmm.yaml]: done

File: .github/workflows/postgresql-12-pmm.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+name: postgresql-12-pmm-integration
+on: pull_request
+
+jobs:
+  build:
+    name: pg-12-pgsm-pmm-integration-test
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (puneet0191) [reply .github/workflows/postgresql-12-pmm.yaml]: done

File: .github/workflows/postgresql-14-pmm.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: postgresql-14-pmm-integration
+on: pull_request
+
+jobs:
+  build:
+    name: pg-14-pgsm-pmm-integration-test
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (puneet0191) [reply .github/workflows/postgresql-14-pmm.yaml]: done

File: .github/workflows/postgresql-13-pmm.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+name: postgresql-13-pmm-integration
+on: pull_request
+
+jobs:
+  build:
+    name: pg-13-pgsm-pmm-integration-test
+    runs-on: ubuntu-20.04
```
[the remark this answers is not in the document]
Comment (puneet0191) [reply .github/workflows/postgresql-13-pmm.yaml]: done

**Comment by capri1989** [comment]
> I see, in repos of core team we have button "merge and squash" in github and you need only click on it, everything else is done by github (merge and squash). So no extra effort, maybe we can configure this repo in same way? What do you think @capri1989 ?

It is enabled on this repository, so there is no technical problem with it and we can always do that in case the commits weren't squashed. The reason why I like this to be done by the author of a PR is, that the person merging the commit, needs to properly define the "single" commit message and might need to go through commit history etc. In this case I'm always inclined for the principle "Don't put the burden on the reader".

**Pull request closed without merging**